.github/workflows/build.yml

```diff
@@ -484,6 +484,31 @@ jobs:
     - name: Tests
       run: xvfb-run make test
 
+  build_tsan_free_threading:
+    name: 'Thread sanitizer (free-threading)'
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
+    needs: check_source
+    if: needs.check_source.outputs.run_tests == 'true'
+    env:
+      TSAN_OPTIONS: "suppressions=${GITHUB_WORKSPACE}/Tools/tsan/supressions.txt"
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install Dependencies
+      run: sudo ./.github/workflows/posix-deps-apt.sh
+    - name: Set up GCC-10 for TSAN
```

Comment:
Why gcc 10 specifically? Perhaps you can simply use a more recent Ubuntu version than 20.04?

Reply:
I just try to match the gcc version with ASAN test.

Reply:
Okay I will try to use the default version.

Reply:
The latest gcc is needed for solving several build issues with the TSAN option. :(